io.openems.edge.meter.tibber/src/io/openems/edge/meter/tibber/pulse/MeterTibberPulseImpl.java

```diff
@@ -0,0 +1,228 @@
+import io.openems.edge.common.component.AbstractOpenemsComponent;
+import io.openems.edge.common.component.OpenemsComponent;
+import io.openems.edge.common.event.EdgeEventConstants;
+import io.openems.edge.meter.api.ElectricityMeter;
+import io.openems.edge.meter.api.MeterType;
+import io.openems.edge.meter.tibber.pulse.decoder.SmlDecoder;
+import io.openems.edge.meter.tibber.pulse.decoder.SmlMeterData;
+import io.openems.edge.timedata.api.Timedata;
+import io.openems.edge.timedata.api.utils.CalculateEnergyFromPower;
+import io.openems.edge.timedata.api.TimedataProvider;
+
+@Designate(ocd = Config.class, factory = true)
+@Component(//
+		name = "Meter.Tibber.Pulse", //
+		immediate = true, //
+		configurationPolicy = ConfigurationPolicy.REQUIRE//
+)
+@EventTopics({ //
+		EdgeEventConstants.TOPIC_CYCLE_EXECUTE_WRITE, //
+		EdgeEventConstants.TOPIC_CYCLE_AFTER_PROCESS_IMAGE//
+})
+public class MeterTibberPulseImpl extends AbstractOpenemsComponent
+		implements MeterTibberPulse, ElectricityMeter, OpenemsComponent, EventHandler, TimedataProvider {
+
+	private final Logger log = LoggerFactory.getLogger(MeterTibberPulseImpl.class);
+
+	@Reference(policy = ReferencePolicy.DYNAMIC, //
+			policyOption = ReferencePolicyOption.GREEDY, //
+			cardinality = ReferenceCardinality.OPTIONAL //
+	)
+	private volatile Timedata timedata;
+
+	private final CalculateEnergyFromPower calculateProductionEnergy = new CalculateEnergyFromPower(this,
+			ElectricityMeter.ChannelId.ACTIVE_PRODUCTION_ENERGY);
+	private final CalculateEnergyFromPower calculateConsumptionEnergy = new CalculateEnergyFromPower(this,
+			ElectricityMeter.ChannelId.ACTIVE_CONSUMPTION_ENERGY);
+
+	private MeterType meterType = null;
+	private String baseUrl;
+	private String credentials;
+	private String encodedAuth;
+
+	@Reference
+	private BridgeHttpFactory httpBridgeFactory;
+	private BridgeHttp httpBridge;
+
+	public MeterTibberPulseImpl() {
+		super(//
+				OpenemsComponent.ChannelId.values(), //
+				ElectricityMeter.ChannelId.values(), //
+				MeterTibberPulse.ChannelId.values());
+	}
+
+	@Activate
+	private void activate(ComponentContext context, Config config) {
+		super.activate(context, config.id(), config.alias(), config.enabled());
+		this.meterType = config.type();
+		this.httpBridge = this.httpBridgeFactory.get();
+
+		this.baseUrl = "http://" + config.ip();
+		this.credentials = "admin:" + config.password();
+		this.encodedAuth = Base64.getEncoder().encodeToString(this.credentials.getBytes());
+
+		if (!this.isEnabled()) {
+			return;
+		}
+
+	}
+
+	@Override
+	@Deactivate
+	protected void deactivate() {
+		super.deactivate();
+		this.httpBridgeFactory.unget(this.httpBridge);
+		this.httpBridge = null;
+	}
+
+	@Override
+	public String debugLog() {
+		var b = new StringBuilder();
+		b.append(this.getActivePowerChannel().value().asString());
+		return b.toString();
+	}
+
+	@Override
+	public void handleEvent(Event event) {
+		this.generateGetRequest();
```

Comment:
für das wäre es besser wenn man wie die anderen methoden eine `subscribeCycle `methode hat in der `BridgeHttp`